ksql-engine/src/main/java/io/confluent/ksql/structured/SchemaKStream.java

```diff
@@ -271,35 +272,28 @@
     );
   }
 
+  @SuppressWarnings("unchecked")
   public SchemaKStream<K> into(
       final String kafkaTopicName,
-      final Serde<GenericRow> topicValueSerDe,
       final LogicalSchema outputSchema,
       final ValueFormat valueFormat,
       final Set<SerdeOption> options,
-      final Set<Integer> rowkeyIndexes,
-      final QueryContext.Stacker contextStacker
+      final QueryContext.Stacker contextStacker,
+      final KsqlQueryBuilder queryBuilder
   ) {
-    kstream
-        .mapValues(row -> {
-          if (row == null) {
-            return null;
-          }
-          final List<Object> columns = new ArrayList<>();
-          for (int i = 0; i < row.getColumns().size(); i++) {
-            if (!rowkeyIndexes.contains(i)) {
-              columns.add(row.getColumns().get(i));
-            }
-          }
-          return new GenericRow(columns);
-        }).to(kafkaTopicName, Produced.with(keySerde, topicValueSerDe));
-    final ExecutionStep<KStream<K, GenericRow>> step = ExecutionStepFactory.streamSink(
+    final StreamSink<K> step = ExecutionStepFactory.streamSink(
         contextStacker,
         outputSchema,
         Formats.of(keyFormat, valueFormat, options),
         sourceStep,
         kafkaTopicName
     );
+    StreamSinkBuilder.build(
+        kstream,
+        step,
+        (fmt, schema, ctx) -> keySerde,
```

Comment:
Once we build the streams app from a visitor, we'll actually get a factory from the source node. For now, we can just pass in the serde we have.

Reply:
I actually think that the Logical model shouldn't know anything about serializers.  The source node should know about schemas, format and serde options only.  The physical builder should be able to convert these to a serde, or a serde factory.

Reply:
That's what I mean. By source node I mean the "source execution step of this step". So the physical builder will do something like:
```
class KStreamAndKeySerdeFactory<K> {
    KStream<K, GenericRow> kstream;
    KeySerdeFactory<K> factory;
}

KStreamAndKeySerdeFactory<K> visitStreamSink(final StreamSink<K> streamSink) {
    final KStreamAndKeySerdeFactory<K> k = streamSink.getSource().accept(this);
    StreamSinkBuilder.build(streamSink.kstream, streamSink, streamSink.factory, ...);
    ...
}
```